### PR TITLE
Test for ipahealtcheck.ipa.idns check when integrated DNS is setup

### PR DESCRIPTION
**Pull request by menonsudhir**
This testcase ensures that the ipahealthcheck.ipa.dns check displays correct result
when ipa master/replica is setup with integrated dns.

### CODE REVIEW

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 1]
```diff
@@ -252,9 +250,7 @@ def test_source_ipahealthcheck_meta_services_check_sssd(self):
         """
         self.master.run_command(["systemctl", "stop", "sssd"])
         returncode, data = run_healthcheck(
-            self.master,
-            "ipahealthcheck.meta.services",
-            "sssd",
+            self.master, "ipahealthcheck.meta.services", "sssd",
         )
```
Comment (rcritten) [inline ipatests/test_integration/test_ipahealthcheck.py]: Is there a reason this unrelated reformatting is included?

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 2]
```diff
@@ -453,10 +430,51 @@ def test_ipa_healthcheck_log_rotate_file_exist_issue35(self):
...
+            self.replicas[0].ip,
+        ]
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.idns",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] in SRV_MASTER_RECORDS
+            assert check["kw"]["key"] in SRV_REPLICA_RECORDS
```
Comment (rcritten) [inline ipatests/test_integration/test_ipahealthcheck.py]: It may work if you change to an or instead. Any given key will only be for one of the records.

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 1 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: @rcritten actually this was done by the tool i had used to format, will revert the changes.

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 2 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: @rcritten have made the change

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 3]
```diff
@@ -453,10 +432,63 @@ def test_ipa_healthcheck_log_rotate_file_exist_issue35(self):
...
+            self.replicas[0].hostname + ".",
+            "_kerberos-replica._tcp." + self.replicas[0].domain.name +
+            ".:" + self.replicas[0].hostname + ".",
+            "_kerberos-replica._udp." + self.replicas[0].domain.name +
+            ".:" + self.replicas[0].hostname + ".",
+            "_kpasswd._tcp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            "_kpasswd._udp." + self.replicas[0].domain.name + ".:" +
+            self.replicas[0].hostname + ".",
+            '"' + self.replicas[0].domain.name.upper() + '"',
```
Comment (rcritten) [inline ipatests/test_integration/test_ipahealthcheck.py]: The full key is \"EXAMPLE.TEST\" so you need those escape strings in there.

Also please use domain.realm.upper instead of domain.name.upper. They just happen to match for your case.

**Comment by rcritten** [comment]
This will fix it. It passes on my local systems:

```
diff --git a/ipatests/test_integration/test_ipahealthcheck.py b/ipatests/test
_integration/test_ipahealthcheck.py
index 04f507c8b..60664ae10 100644
--- a/ipatests/test_integration/test_ipahealthcheck.py
+++ b/ipatests/test_integration/test_ipahealthcheck.py
@@ -472,7 +472,8 @@ class TestIpaHealthCheck(IntegrationTest):
             "_kpasswd._udp." + self.master.domain.name + ".:" +
             self.master.hostname + ".",
             "\"" + self.master.domain.realm.upper() + "\"",
-            "\"" + self.master.ip + "\""
+            self.master.ip,
+            self.replicas[0].ip
         ]
         returncode, data = run_healthcheck(
             self.master, "ipahealthcheck.ipa.idns",

```

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 4]
```diff
@@ -21,7 +21,7 @@
 HEALTHCHECK_LOG_ROTATE_CONF = "/etc/logrotate.d/ipahealthcheck"
 HEALTHCHECK_LOG_DIR = "/var/log/ipa/healthcheck"
 HEALTHCHECK_OUTPUT_FILE = "/tmp/output.json"
-HEALTHCHECK_PKG = ['freeipa-healthcheck']
+HEALTHCHECK_PKG = ["freeipa-healthcheck"]
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py]: Can you update it  to  `HEALTHCHECK_PKG = ["*ipa-healthcheck"]` ?

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 5]
```diff
@@ -333,17 +323,14 @@ def test_source_ipahealthcheck_ipa_host_check_ipahostkeytab(self):
...
         )
-        dirsrv_ipactl_status = 'Directory Service: STOPPED'
+        dirsrv_ipactl_status = "Directory Service: STOPPED"
         api.env.realm = self.master.domain.name
         serverid = (realm_to_serverid(api.env.realm)).upper()
         dirsrv_service = "dirsrv@%s.service" % serverid
         self.master.run_command(["systemctl", "stop", dirsrv_service])
-        result = self.master.run_command(
-            ["ipactl", "status"])
+        result = self.master.run_command(["ipactl", "status"])
```
Comment (mrizwan93) [inline ipatests/test_integration/test_ipahealthcheck.py]: Are we using `result` anywhere else?

Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: @mrizwan93 removed the result since we are not using it.

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 4 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: @mrizwan93 have made this change as well to ensure it works for other distros.

**Comment by rcritten** [comment]
The patch looks ok to me but then unrelated changes need to be backed out.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 6]
```diff
@@ -423,13 +426,11 @@ def test_ipa_healthcheck_dna_plugin_returns_warning_pagure_issue_60(self):
...
         tasks.user_add(
-            self.replicas[0], 'ipauser1', first='Test', last='User',
+            self.replicas[0], "ipauser1", first="Test", last="User",
         )
         # Now run the ipa-healthcheck command again
         returncode, data = run_healthcheck(
-            self.replicas[0],
-            "ipahealthcheck.ipa.dna",
-            "IPADNARangeCheck",
+            self.replicas[0], "ipahealthcheck.ipa.dna", "IPADNARangeCheck",
```
Comment (rcritten) [inline ipatests/test_integration/test_ipahealthcheck.py]: fixing formatting not relevant here.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 7]
```diff
@@ -423,13 +426,11 @@ def test_ipa_healthcheck_dna_plugin_returns_warning_pagure_issue_60(self):
         # DNA configuration.
         tasks.kinit_admin(self.replicas[0])
         tasks.user_add(
-            self.replicas[0], 'ipauser1', first='Test', last='User',
+            self.replicas[0], "ipauser1", first="Test", last="User",
```
Comment (rcritten) [inline ipatests/test_integration/test_ipahealthcheck.py]: Changing quotes not needed.

File: ipatests/test_integration/test_ipahealthcheck.py [hunk 8]
```diff
@@ -285,17 +284,21 @@ def test_source_ipahealthcheck_dogtag_ca_dogtagcertsconfigcheck(self):
...
             "ipahealthcheck.dogtag.ca",
             "DogtagCertsConfigCheck",
         )
         assert returncode == 1
         assert data[0]["result"] == "CRITICAL"
-        self.master.run_command(["mv", TOMCAT_CFG + ".old", TOMCAT_CFG])
+        self.master.run_command(
+            ["mv", paths.CA_CS_CFG_PATH + ".old", paths.CA_CS_CFG_PATH]
+        )
         self.master.run_command(["ipactl", "restart"])
```
Comment (rcritten) [inline ipatests/test_integration/test_ipahealthcheck.py]: Use a constant is probably a good idea but should be done in a separate patch IMHO.

**Comment by rcritten** [comment]
Can you add more details to the commit message? A BZ or Pagure link?

Additional text like:

Add test to compare the expected service records vs what is returned by healthcheck.

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 8 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: @rcritten ok will remove this and add in a seperate patch.

File: ipatests/test_integration/test_ipahealthcheck.py (hunk 6 above)
Comment (menonsudhir) [reply ipatests/test_integration/test_ipahealthcheck.py]: @rcritten the changes have been reverted.

**Comment by menonsudhir** [comment]
> Can you add more details to the commit message? A BZ or Pagure link?
> 
> Additional text like:
> 
> Add test to compare the expected service records vs what is returned by healthcheck.

@rcritten have added the description and bz link.

**Comment by rcritten** [comment]
Perfect, please remove the temp commit.

**Comment by rcritten** [comment]
master:

* fd9f1b3d5bf35f8286505d1ed0971a42349a476b Test for ipahealthcheck.ipa.idns check when integrated DNS is setup
